integreat_compass/cms/models/offers/offer.py

```diff
@@ -0,0 +1,92 @@
+from django.db import models
+from django.utils.functional import cached_property
+from django.utils.translation import gettext_lazy as _
+
+from ...constants import offer_group_types, offer_mode_types, offer_version_states
+from ..abstract_base_model import AbstractBaseModel
+from ..interactions.comment import Comment
+from ..users.user import User
+from .tag import Tag
+
+
+class Offer(AbstractBaseModel):
+    """
+    Data model representing an Offer.
+    """
+
+    creator = models.ForeignKey(User, on_delete=models.CASCADE)
+    organization = models.OneToOneField("Organization", on_delete=models.CASCADE)
+    offer_contact = models.OneToOneField("Contact", on_delete=models.CASCADE)
+    location = models.OneToOneField("Location", on_delete=models.CASCADE)
+    tags = models.ManyToManyField(Tag)
+    group_type = models.CharField(
+        choices=offer_group_types.CHOICES,
+        null=False,
+        default=offer_group_types.GROUP,
+        verbose_name=_("group type"),
+        help_text=_("Select in what group sizes lessons are offered"),
+    )
+    mode_type = models.CharField(
+        choices=offer_mode_types.CHOICES,
+        null=False,
+        default=offer_mode_types.IN_PERSON,
+        verbose_name=_("lesson mode"),
+        help_text=_("Select in what mode lessons are offered"),
+    )
+
+    @cached_property
+    def public_version(self):
```

Comment:
```suggestion
    def approved_version(self):
```

Reply:
Hm... There can be many approved versions. And I think it makes intuitive sense to call the version that should be displayed to visitors the public version.

Reply:
Ok, nevermind :+1: